great_expectations/expectations/model_field_types.py

```diff
@@ -22,11 +22,13 @@
 ]
 
 ValueSetField = Annotated[
-    Union[Sequence, set, SuiteParameterDict],
-    pydantic.Field(
+    Union[List, SuiteParameterDict],
+    Field(
         title="Value Set",
         description=VALUE_SET_DESCRIPTION,
+        # Add type constraint to exclude strings and bytes
         schema_overrides={
+            "not": {"anyOf": [{"type": "string"}, {"type": "string", "format": "binary"}]},
```

Comment:
for reviewers - this prevents string and bytestring from coercing to a list. Should we skip this change and instead have an expectation validation error if someone passes a string instead of expected iterables (which is unlikely but possible)?
I am also not sure if we want to update schemas - can remove this addition and undo schema changes

Reply:
Does this actually prevent the type coercion when using the API, or does it just change the schema? If it just changes the schema, I don't think it's necessary. You could raise an error in the validator if the wrong type is passed. This could be a more widespread problem so I wouldn't spend too much time thinking about it.

Reply:
I guess that's what my question was. I wouldn't expect it to raise a validation error if you are just updating the schema. Are you sure if you remove the schema changes that it will fail to raise the error?